### ISSUE TITLE
Packaging: is casadi required ?

### ISSUE THREAD

**Issue by nim65s** [Simple-Robotics/proxsuite-nlp#51]
## Bug description

On a system without casadi, running unittests fail with:
```python
15/18 Test #15: test-py-1d-funs...................***Failed    0.23 sec
Traceback (most recent call last):
  File "…/proxsuite-nlp-0.3.1/tests/python/test_1d_funs.py", line 3, in <module>
    from proxsuite_nlp.casadi_utils import CasadiFunction
  File "…/proxsuite-nlp-0.3.1/bindings/python/proxsuite_nlp/casadi_utils.py", line 4, in <module>
    import casadi
ModuleNotFoundError: No module named 'casadi'
```

## Expected behavior

We should either declare this dependency in CMake, or make it really optional

**Comment by jcarpent** [comment]
Solved via #65

**Comment by ManifoldFR** [comment]
I also fixed the test that was using casadi before in #81
